core/domain/prod_validators.py

```diff
@@ -4015,5 +4015,10 @@ def _get_external_id_relationships(cls, item):
                 'user_settings_ids',
                 user_models.UserSettingsModel,
                 [item.user_id]
+            ),
+            base_model_validators.ExternalModelFetcherDetails(
+                'user_auth_details_ids',
+                user_models.UserAuthDetailsModel,
+                [item.user_id]
```

Comment:
Are we validating that the item.gae_id matches any UserAuthDetailsModel.gae_id anywhere? (I think yes, but just wanted to make sure!)

Reply:
No, but added the validation to `UserAuthDetailsModelValidator`.